pkg/registry/apis/provisioning/files.go

```diff
@@ -76,14 +76,19 @@ func (s *filesConnector) Connect(ctx context.Context, name string, opts runtime.
 		logger := logger.With("url", r.URL.Path, "ref", ref, "message", message)
 		ctx := logging.Context(r.Context(), logger)
 
-		filePath, err := ExtractFilePath(r.URL.Path, fmt.Sprintf("/%s/files/", name))
+		filePath, err := pathAfterPrefix(r.URL.Path, fmt.Sprintf("/%s/files/", name))
```

Comment:
```suggestion
		filePath, err := pathAfterPrefix(r.URL.Path, fmt.Sprintf("/%s/files", name))
```

If we want the `/` prefix here, we wouldn't catch file directory listings that don't include the trailing slash (AKA in tests with the K8s HTTP client, as it strips trailing slashes...).